packages/lib/models/Revision.ts

```diff
@@ -305,43 +308,82 @@ export default class Revision extends BaseItem {
 		// and modify that revision into a "merged" one.
 
 		const cutOffDate = Date.now() - ttl;
-		const revisions = await this.modelSelectAll('SELECT * FROM revisions WHERE item_updated_time < ? ORDER BY item_updated_time DESC', [cutOffDate]);
-		const doneItems: Record<string, boolean> = {};
+		const allOldRevisions: RevisionEntity[] = await this.modelSelectAll(
+			'SELECT * FROM revisions WHERE item_updated_time < ? ORDER BY item_updated_time DESC',
+			[cutOffDate],
+		);
+
+		const itemIdToOldRevisions = new Map<string, RevisionEntity[]>();
+		for (const rev of allOldRevisions) {
+			const itemId = rev.item_id;
+			if (!itemIdToOldRevisions.has(itemId)) {
+				itemIdToOldRevisions.set(itemId, []);
+			}
+			itemIdToOldRevisions.get(itemId).push(rev);
+		}
 
-		for (const rev of revisions) {
-			const doneKey = `${rev.item_type}_${rev.item_id}`;
-			if (doneItems[doneKey]) continue;
+		const deleteOldRevisionsForItem = async (itemType: ModelType, itemId: string, oldRevisions: RevisionEntity[]) => {
+			const item = await BaseItem.loadItemById(itemId);
```

Comment:
- [ ] Verify that this doesn't fail if:
      1. A note is created.
      2. The note is permanently deleted.
      3. The note's revisions are processed.